src/protocol/sort/generate_permutation.rs

```diff
@@ -110,27 +130,32 @@ where
     let mut composed_less_significant_bits_permutation = bit_0_permutation;
     for bit_num in 1..num_bits {
         let ctx_bit = ctx.narrow(&Sort(bit_num));
-        let revealed_and_random_permutations = shuffle_and_reveal_permutation(
+        let reveal_permutation_output = shuffle_and_reveal_permutation(
             ctx_bit.narrow(&ShuffleRevealPermutation),
             input_len,
             composed_less_significant_bits_permutation,
+            None,
         )
         .await?;
 
         let bit_i_sorted_by_less_significant_bits = secureapplyinv(
             ctx_bit.narrow(&ApplyInv),
             sort_keys[bit_num as usize].clone(),
             (
-                revealed_and_random_permutations
+                reveal_permutation_output
+                    .reveal_and_random_permutation
                     .randoms_for_shuffle
                     .0
                     .as_slice(),
-                revealed_and_random_permutations
+                reveal_permutation_output
+                    .reveal_and_random_permutation
                     .randoms_for_shuffle
                     .1
                     .as_slice(),
             ),
-            &revealed_and_random_permutations.revealed,
+            &reveal_permutation_output
+                .reveal_and_random_permutation
+                .revealed,
```

Comment:
This really looks like it should be a method on `reveal_permutation_output` given how much rummaging around it does in the internals of that object.

Reply:
This is going away thanks to the new downgrade trait implementation!